src/core/bau_inputs.jl

```diff
@@ -164,15 +175,134 @@ function BAUInputs(p::REoptInputs)
         p.pbi_max_kw, 
         p.pbi_benefit_per_kwh,
         boiler_efficiency,
+        tech_renewable_energy_pct, 
+        tech_emissions_factors_CO2, 
+        tech_emissions_factors_NOx, 
+        tech_emissions_factors_SO2, 
+        tech_emissions_factors_PM25,
         p.techs_operating_reserve_req_pct
     )
 end
 
 
+"""
+    update_bau_emissions_inputs(s::AbstractScenario, bau_emissions_lb_CO2_per_year, bau_emissions_lb_NOx_per_year, 
+        bau_emissions_lb_SO2_per_year, bau_emissions_lb_PM25_per_year, bau_grid_emissions_lb_CO2_per_year, 
+        bau_grid_emissions_lb_NOx_per_year, bau_grid_emissions_lb_SO2_per_year, bau_grid_emissions_lb_PM25_per_year)
+
+Update the `bau_(grid_)emissions_` values in s.site
+"""
+function update_bau_emissions_inputs(s::AbstractScenario, bau_emissions_lb_CO2_per_year, bau_emissions_lb_NOx_per_year, 
+            bau_emissions_lb_SO2_per_year, bau_emissions_lb_PM25_per_year, bau_grid_emissions_lb_CO2_per_year, 
+            bau_grid_emissions_lb_NOx_per_year, bau_grid_emissions_lb_SO2_per_year, bau_grid_emissions_lb_PM25_per_year)
+    
+    s.site.bau_emissions_lb_CO2_per_year = bau_emissions_lb_CO2_per_year
+    s.site.bau_emissions_lb_NOx_per_year = bau_emissions_lb_NOx_per_year
+    s.site.bau_emissions_lb_SO2_per_year = bau_emissions_lb_SO2_per_year
+    s.site.bau_emissions_lb_PM25_per_year = bau_emissions_lb_PM25_per_year
+
+    s.site.bau_grid_emissions_lb_CO2_per_year = bau_grid_emissions_lb_CO2_per_year
+    s.site.bau_grid_emissions_lb_NOx_per_year = bau_grid_emissions_lb_NOx_per_year
+    s.site.bau_grid_emissions_lb_SO2_per_year = bau_grid_emissions_lb_SO2_per_year
+    s.site.bau_grid_emissions_lb_PM25_per_year = bau_grid_emissions_lb_PM25_per_year
+end
+
+"""
+    setup_bau_emissions_inputs(p::REoptInputs, s_bau::BAUScenario, generator_fuel_use_gal::Float64)
+
+Pre-processing of the BAU emissions to use in determining emissions reductions in the optimal case
+Include BAU grid emissions, existing backup generator emissions, boiler emissions
+Update the `bau_(grid_)emissions_` values in s.site and s_bau.site
+
+Note if existing generation does not sustain a simulated outage, the BAU load and therefore emissions
```

Comment:
Does this apply for multi-outage situations or just single outages?

Reply:
done in 9f9c77a

Reply:
done in commit 9f9c77a